pkg/encryptedstore/secrets_store.go

```diff
@@ -65,18 +65,21 @@ func (g *GenericEncryptedStore) getKey(name string) string {
 	return g.prefix + name
 }
 
-func (g *GenericEncryptedStore) Set(name string, data map[string]string) error {
-	return g.set(name, data)
+func (g *GenericEncryptedStore) Set(name string, data map[string]string, owner *metav1.OwnerReference) error {
+	return g.set(name, data, owner)
 }
 
-func (g *GenericEncryptedStore) set(name string, data map[string]string) error {
+func (g *GenericEncryptedStore) set(name string, data map[string]string, owner *metav1.OwnerReference) error {
 	logrus.Debugf("[GenericEncryptedStore]: set secret called for %v", g.getKey(name))
 	sec, err := g.secretLister.Get(g.namespace, g.getKey(name))
 	if errors.IsNotFound(err) {
 		logrus.Debugf("[GenericEncryptedStore]: Creating secret for %v", g.getKey(name))
 		sec = &corev1.Secret{}
 		sec.Name = g.getKey(name)
 		sec.StringData = data
+		if owner != nil {
+			sec.SetOwnerReferences([]metav1.OwnerReference{*owner})
```

Comment:
@jiaqiluo @jakefhyde Do you see issues with us also doing this for update so logic also applies for existing custom clusters?

Reply:
No I don't see any issues there. @pratikjagrut WDYT?